### DISCUSSION TITLE
Creating a library to translate (i18n) default error messages.

### DISCUSSION THREAD

**Discussion by aiji42** [colinhacks/zod#1329] [Show and tell]
I am developing a library to translate zod's default error messages into another language with `i18next` and to change the format of the messages.
https://github.com/aiji42/zod-i18n

I have a few requests for you. If you agree with the purpose of this library, please help us create translation files in your language. (Currently only English and Japanese translations are available.)
Also, if you would like to try out the library in your project, we would like you to evaluate its usefulness in a real use case.

Your contribution is appreciated 🙏.

**Comment by nullndr** [comment]
Just to ask but why are you translating zod defaults message?

I don't think they suits for real use cases, like your [vercel](https://zod-i18n.vercel.app/)

![zod18n](https://user-images.githubusercontent.com/62137266/184169446-73576313-9178-4da9-8bc2-e51c80545dc8.png)

Like in this example I don't see `String must contain at least 5 character(s)` very well, should be something like `Username should be at least 5 characters` or something like that.

**Reply by aiji42** [reply]
> Just to ask but why are you translating zod defaults message?

For users of non-English languages, English error messages are not suitable and need to be translated. Currently, translating messages requires defining a message option for each schema, which is redundant.

> Like in this example I don't see String must contain at least 5 character(s) very well, should be something like Username should be at least 5 characters or something like that.

I think it is still necessary to define a message option in each schema for those error messages that are not easily understood by the default error messages.   
`username: z.string().min(5, { message: 'Username should be at least 5 characters' })`

I think one way to do this would be to customize the translation files. 
https://github.com/aiji42/zod-i18n/blob/main/packages/core/locales/en/zod.json#L31
`"inclusive": "Please input at least {{minimum}} character(s).",`

Anyway, these two things are the purpose of this library.
- Translation for non-English speaking users
- Customizing translation files to change and batch manage error message formats

This library does not cover the issue of making error messages easier to understand for individual schemas.
Since there is no translation requirement for English users, the benefit of this library is limited only to the customization of error message formatting.
